### PR TITLE
fix: use Symbol.for() for MIDDLEWARE_CONTEXT 

### PR DESCRIPTION
**Pull request by mschnee**
Signed-off-by: Matthew Schnee <i.am@matthew.engineer>

The middleware chain is bound in context by using the Symbol `import { MIDDLEWARE_CONTEXT } from '@loopback/rest`;
However, `@loopback/graphql` was creating a different `MIDDLEWARE_CONTEXT`, and `LoopBackCountainer` was using a default context instead, because `Symbol.for('name') !== Symbol('name')`.  This could be caused due to order of imports.

## Use Case
Given a system that already has a lot of services expecting `@inject(SecurityBindings.CURRENT_USER)`, and a desire to use inject these services and providers into a `GraphQLResolver`, we need to verify and bind the CurrentUser ourselves:
```ts
// in Application.ts
this.bind(GraphQLBindings.GRAPHQL_CONTEXT_RESOLVER).to(async (graphQLCtx) => {
  // we need to get the request Context from the given express request first,
  // so that we can set the CURRENT_USER for services to inject
  const reqCtx = (graphQLCtx?.req as any)?.[MIDDLEWARE_CONTEXT];

  const tokenService = await this.get<MySpecificTokenService>('services.MySpecificTokenService');
  if (graphQLCtx.req.headers?.authorization?.startsWith('Bearer ')) {
    const token = graphQLCtx.req.headers.authorization.replace('Bearer ', '');
    const currentUser = await tokenService.verifyToken(token, graphQLCtx.req);
    reqCtx?.bind(AuthenticationBindings.CURRENT_USER).to(currentUser);
  }
  return graphQLCtx;
});
```

The effect of this code is that we use `MySpecificTokenService` as the one-and-only token resolver.  This is a brute-force method to ensure that CurrentUser is available when LoopBackContainer gets the resolver in `LoopbackContainer.get()`

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: extensions/graphql/src/graphql.container.ts [hunk 1]
```diff
@@ -12,13 +12,12 @@ import {
...
   filterByServiceInterface,
 } from '@loopback/core';
+import {MIDDLEWARE_CONTEXT} from '@loopback/rest';
 import {ExpressContext} from 'apollo-server-express/dist/ApolloServer';
 import debugFactory from 'debug';
 import {ContainerType, ResolverData} from 'type-graphql';
 import {GraphQLBindings, GraphQLTags} from './keys';
-
 const debug = debugFactory('loopback:graphql:container');
-const MIDDLEWARE_CONTEXT = Symbol.for('loopback.middleware.context');
```
Comment (raymondfeng) [inline extensions/graphql/src/graphql.container.ts:21]: I think it's by design that we don't want to have hard dependency on `@loopback/rest`.

`Symbol.for('loopback.middleware.context')` resolves a global symbol by name if it exists. In our case, there are two possibilities:

1. There are different versions of `@loopback/express` modules in the same app dependency tree
2. The module loading order matters. (If graphql.container.js is loaded before packages/express/src/types.ts (line 330). Try to change this line to be `Symbol.for` as well may fix the issue.

Comment (mschnee) [reply extensions/graphql/src/graphql.container.ts:21]: Okay, testing that out :)

Comment (mschnee) [reply extensions/graphql/src/graphql.container.ts]: This works, I am pushing that change now :)

File: packages/express/src/types.ts [hunk 2]
```diff
@@ -327,7 +327,7 @@ export interface ExpressMiddlewareFactory<C> {
 /**
  * A symbol to store `MiddlewareContext` on the request object
  */
-export const MIDDLEWARE_CONTEXT = Symbol('loopback.middleware.context');
+export const MIDDLEWARE_CONTEXT = Symbol.for('loopback.middleware.context');
```
Comment (raymondfeng) [inline packages/express/src/types.ts:330]: Please add a comment mentioning that the symbol may be referenced by name before it is created. Using Symbol.for removes the module loading order sensitivity.

Comment (mschnee) [reply packages/express/src/types.ts:330]: Done!

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**